network/src/counters.rs

```diff
@@ -66,47 +60,111 @@ pub static LIBRA_NETWORK_RPC_MESSAGES: Lazy<IntCounterVec> = Lazy::new(|| {
     register_int_counter_vec!(
         "libra_network_rpc_messages",
         "Number of RPC messages",
-        &["type", "state"]
+        &["role_type", "network_id", "self_peer_id", "type", "state"]
     )
     .unwrap()
 });
 
+pub fn rpc_messages(
+    network_context: &NetworkContext,
+    type_label: &'static str,
+    state_label: &'static str,
+) -> IntCounter {
+    LIBRA_NETWORK_RPC_MESSAGES.with_label_values(&[
+        network_context.role().as_str(),
+        network_context.network_id().as_cow_str().as_ref(),
+        network_context.peer_id_short_str(),
+        type_label,
+        state_label,
+    ])
+}
+
+// TODO(philiphayes): the default histogram buckets don't make sense for measuring
+// message sizes; we need to set these explicitly to something reasonable...
+
 pub static LIBRA_NETWORK_RPC_BYTES: Lazy<HistogramVec> = Lazy::new(|| {
     register_histogram_vec!(
         "libra_network_rpc_bytes",
         "Number of RPC bytes transferred",
-        &["type", "state"]
+        &["role_type", "network_id", "self_peer_id", "type", "state"]
     )
     .unwrap()
 });
 
+pub fn rpc_bytes(
+    network_context: &NetworkContext,
+    type_label: &'static str,
+    state_label: &'static str,
+) -> Histogram {
+    LIBRA_NETWORK_RPC_BYTES.with_label_values(&[
+        network_context.role().as_str(),
+        network_context.network_id().as_cow_str().as_ref(),
+        network_context.peer_id_short_str(),
+        type_label,
+        state_label,
+    ])
+}
+
+// TODO(philiphayes): specify that this is outbound rpc only
+// TODO(philiphayes): somehow get per-peer latency metrics without using a
+// separate peer_id label ==> cardinality explosion.
+
 pub static LIBRA_NETWORK_RPC_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
     register_histogram_vec!(
         "libra_network_rpc_latency_seconds",
         "RPC request latency in seconds",
-        &["type", "protocol_id", "peer_id"]
+        &["role_type", "network_id", "self_peer_id"]
     )
     .unwrap()
 });
 
+pub fn rpc_latency(network_context: &NetworkContext) -> Histogram {
+    LIBRA_NETWORK_RPC_LATENCY.with_label_values(&[
+        network_context.role().as_str(),
+        network_context.network_id().as_cow_str().as_ref(),
+        network_context.peer_id_short_str(),
+    ])
+}
+
 pub static LIBRA_NETWORK_DIRECT_SEND_MESSAGES: Lazy<IntCounterVec> = Lazy::new(|| {
     register_int_counter_vec!(
         "libra_network_direct_send_messages",
         "Number of direct send messages",
-        &["state"]
+        &["role_type", "network_id", "self_peer_id", "state"]
     )
     .unwrap()
 });
 
+pub fn direct_send_messages(
```

Comment:
again would recommend we just aggregate this ignoring the protocol type and focus on the message type.... though it isn't quite clear to me what we gain with this metric.

Reply:
yep, I removed the protocol type. I would be fine setting the `rpc_bytes` and `direct_send_bytes` to just plain counters